apps/design/backend/src/file_storage_client.ts

```diff
@@ -0,0 +1,71 @@
+// eslint-disable-next-line max-classes-per-file
+import { Buffer } from 'node:buffer';
+import { createReadStream } from 'node:fs';
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import { Readable } from 'node:stream';
+import {
+  GetObjectCommand,
+  PutObjectCommand,
+  S3Client,
+} from '@aws-sdk/client-s3';
+import { assertDefined } from '@votingworks/basics';
+
+import { WORKSPACE } from './globals';
+
+export interface FileStorageClient {
+  readFile: (filePath: string) => Promise<Readable>;
+  writeFile: (filePath: string, contents: Buffer) => Promise<void>;
+}
+
+/**
+ * A file storage client backed by S3, for deployed environments
+ */
+export class S3FileStorageClient {
+  private readonly s3Client: S3Client;
+
+  constructor() {
+    this.s3Client = new S3Client({ region: process.env.AWS_S3_REGION });
+  }
+
+  async readFile(filePath: string): Promise<Readable> {
+    const data = await this.s3Client.send(
+      new GetObjectCommand({
+        Bucket: process.env.AWS_S3_BUCKET_NAME,
+        Key: filePath,
+      })
+    );
+    return (data.Body ?? Readable.from([])) as Readable;
+  }
+
+  async writeFile(filePath: string, contents: Buffer): Promise<void> {
```

Comment:
Because we're not generating language and audio files this ballot production run, the election packages will be reasonably small. So I think it's okay to hold them in memory for writing purposes. Down the road, we'll want to use streaming APIs. This PR does use streaming APIs for election package downloads.